### PR TITLE
Scanbox format updates

### LINKED ISSUE #1299: Suggestion: remove SBX loading code and recommend using sbxreader instead
### Discussed in https://github.com/flatironinstitute/CaImAn/discussions/1298

<div type='discussions-op-text'>

<sup>Originally posted by **ethanbb** March 14, 2024</sup>
Hello,

This is mainly a documentation suggestion. My microscope saves Scanbox files, and I saw CaImAn has support for loading them, so I tried that but got an error because the loading functions are not up-to-date with newer versions of the file format (which has changed several times). The specific error was about inferring the number of channels (my `info` did not have `'channels'`, but rather the newer `'chan'` field).

Looking into it a little more, I realized that the current function doesn't account for a number of features, including:
- Volumetric scanning (optotune)
- Bidirectional scanning
- "Fold lines" (multiple subframes per frame)
- Loading channels other than the first one

I took a few days to implement these features, and I was going to submit a pull request. However, towards the end I realized that the package [sbxreader](https://github.com/jcouto/sbxreader) exists (which suite2p uses for loading). Besides being actively maintained, it's much better than what I wrote because it memory-maps the file - I realized when trying to test my function that it was not going to happen because my 40-minute recording was too large to fit into memory. It also has another function for loading all metadata, which is probably important to almost everyone who would need to load SBX data.

I could continue to tweak what I wrote to use memmap etc., but I think what makes the most sense is to just recommend Scanbox users to install this package and use it to get an np.memmap of their data, then load it into CaImAn using the movie constructor directly. To encourage this (and because it's out of date), it probably makes sense to remove the current SBX loading code.</div>

### PR DESCRIPTION
**Pull request by ethanbb**
# Description

- Updates .sbx loading code to deal with newer versions of the format;
- Adds functions to convert .sbx files and chains thereof to .tifs for more flexibility;
- Adds a function to load the metadata;
- Pulls all the sbx-relevant code into a separate file under `utils`.

Fixes #1299 

# Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

# Has your PR been tested?

Yes, tests for .sbx format added to `tests/test_sbx.py`, using example datafiles added to `testdata`
All tests passed with `caimanmanager test`. `caimanmanager demotest` fails because of #1310.

### CODE REVIEW

File: caiman/base/movies.py [hunk 1]
```diff
@@ -35,6 +35,7 @@
 import caiman.mmapping
 import caiman.summary_images
 import caiman.utils.visualization
+import caiman.utils.sbx_utils
```
Comment (pgunn) [inline caiman/base/movies.py:38]: Can we flip the order to keep these alphabetised?

File: caiman/base/movies.py [hunk 2]
```diff
@@ -1575,10 +1576,11 @@ def load(file_name: Union[str, list[str]],
 
         elif extension == '.sbx':
             logging.debug('sbx')
-            if subindices is not None:
-                return movie(sbxreadskip(file_name[:-4], subindices), fr=fr).astype(outtype)
-            else:
-                return movie(sbxread(file_name[:-4], k=0, n_frames=np.inf), fr=fr).astype(outtype)
+            meta_data = caiman.utils.sbx_utils.sbx_meta_data(file_name[:-4])
```
Comment (pgunn) [inline caiman/base/movies.py:1579]: We have a lot of code throughout the codebase that does this; eventually we'll want to convert them all to use os.path.splitext() instead of indexing into the string. When I fix them all I'll grep for stuff like this, so it's fine to leave this in (although if you want to adjust it yourself you can)

File: caiman/tests/test_sbx.py [hunk 3]
```diff
@@ -0,0 +1,150 @@
+#!/usr/bin/env python
+
+import os
+
+import numpy as np
+import numpy.testing as npt
```
Comment (pgunn) [inline caiman/tests/test_sbx.py:6]: Can we alphabetise these parts of the imports? (the caiman imports going later is fine

File: caiman/utils/sbx_utils.py [hunk 4]
```diff
@@ -0,0 +1,525 @@
...
+Utility functions for Neurolabware Scanbox files (.sbx)
+"""
+
+import os
+import logging
+from typing import Iterable
+
+import numpy as np
+import scipy
+import tifffile
```
Comment (pgunn) [inline caiman/utils/sbx_utils.py:13]: Can we alphabetise these imports?

**Review by pgunn — COMMENTED** [review]
Very nice. I particularly like your type annotations style; might try out some of that with my own code because of how clean it looks. Left a few little comments.

**Comment by pgunn** [comment]
Looking good, will merge once CI finishes. Thanks!

**Comment by pgunn** [comment]
@ethanbb Ah, I'm fixing CI now and it turns out CI found an issue with the code:

```
  File "/home/runner/work/CaImAn/CaImAn/caiman/base/movies.py", line 37, in <module>
    import caiman.utils.sbx_utils
  File "/home/runner/work/CaImAn/CaImAn/caiman/utils/sbx_utils.py", line 435
    return chunk[:, *np.ix_(*subindices[1:])]
                    ^
SyntaxError: invalid syntax
Error: Process completed with exit code 1.
```

**Comment by pgunn** [comment]
Looks like this is 3.11-specific syntax; is there a chance you can, for now, rephrase it to be compatible back to 3.9?

**Comment by ethanbb** [comment]
Yup, no problem.
